### PR TITLE
aten::var

### PR DESCRIPTION
**Pull request by eaidova**

### CODE REVIEW

File: src/frontends/pytorch/src/op/var.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+namespace op {
+
+OutputVector translate_var(NodeContext& context) {
+    auto data = context.get_input(0);
+    bool unbiased = true;
+    bool keepdims = false;
+    auto num_elements = numel(context, 0);
+    bool keepdim_mean;
+    std::shared_ptr<ov::Node> mean;
+    ov::Output<ov::Node> axes;
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var.cpp]: Please fix code style

Comment (eaidova) [reply src/frontends/pytorch/src/op/var.cpp]: fixed

File: src/frontends/pytorch/src/op/var.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if (context.inputs().size() == 2) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var.cpp:22]: Please add comment about what is the difference between inputs size == 2 and inputs size != 2.

File: src/frontends/pytorch/src/op/var.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            reduced_dims = context.mark_node(std::make_shared<opset8::Gather>(reduced_dims, axes, zero));
+            num_elements = context.mark_node(std::make_shared<opset8::ReduceProd>(reduced_dims, zero, false));
+        }
+        keepdim_mean = context.input_is_none(1) ? false : keepdims;
+    }
+    auto sub_v = context.mark_node(std::make_shared<opset8::Subtract>(data, mean));
+    auto sqr_sub = context.mark_node(std::make_shared<opset8::Multiply>(sub_v, sub_v));
+    auto var = context.mark_node(std::make_shared<opset8::ReduceMean>(sqr_sub, axes, keepdim_mean));
+    if (unbiased) {
+        num_elements = context.mark_node(std::make_shared<opset8::Convert>(num_elements, element::f32));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var.cpp]: Why f32? Shouldn't it be same type as `data`? I think better to use `ConvertLike` here.

File: src/frontends/pytorch/src/op/var.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            auto zero = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {0}));
+            reduced_dims = context.mark_node(std::make_shared<opset8::Gather>(reduced_dims, axes, zero));
+            num_elements = context.mark_node(std::make_shared<opset8::ReduceProd>(reduced_dims, zero, false));
+        }
+        keepdim_mean = context.input_is_none(1) ? false : keepdims;
+    }
+    auto sub_v = context.mark_node(std::make_shared<opset8::Subtract>(data, mean));
+    auto sqr_sub = context.mark_node(std::make_shared<opset8::Multiply>(sub_v, sub_v));
+    auto var = context.mark_node(std::make_shared<opset8::ReduceMean>(sqr_sub, axes, keepdim_mean));
+    if (unbiased) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var.cpp:54]: Please add comment that `unbiased` means Bessel’s correction and is (var * N) / (N - 1)

File: src/frontends/pytorch/src/op/var.cpp [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        num_elements = context.mark_node(std::make_shared<opset8::Convert>(num_elements, element::f32));
+        auto one = context.mark_node(opset8::Constant::create(element::f32, Shape{}, {1}));
+        auto mul = context.mark_node(std::make_shared<opset8::Multiply>(var, num_elements));
+        auto div = context.mark_node(std::make_shared<opset8::Subtract>(num_elements, one));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var.cpp]: `div` name is misleading here

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/op/var.cpp (hunk 5 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/var.cpp]: renamed

File: src/frontends/pytorch/src/op/var.cpp (hunk 4 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/var.cpp:54]: added

File: src/frontends/pytorch/src/op/var.cpp (hunk 3 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/var.cpp]: updated

File: src/frontends/pytorch/src/op/var.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/var.cpp:22]: added comments regarding var operation signature

File: src/frontends/pytorch/src/op/var.cpp [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    std::shared_ptr<ov::Node> mean;
+    ov::Output<ov::Node> axes;
+    if (context.inputs().size() == 2) {
+        // aten::var(input, unbiased)
+        axes = context.mark_node(get_axes_range(context, 0));
+        unbiased = context.const_input<bool>(1);
+        mean = context.mark_node(std::make_shared<opset8::ReduceMean>(data, axes, keepdims));
+        keepdim_mean = keepdims;
+    } else {
+        // aten::var(input, dim, unbiased:bool=None, keepdim:`bool=None)
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var.cpp]:
```suggestion
        // aten::var(input, dim, unbiased:bool=None, keepdim:bool=None)
```

Comment (eaidova) [reply src/frontends/pytorch/src/op/var.cpp]: already applied

**Review by mvafin — APPROVED** [review]

**Pull request merged**